pkg/controller/statefulset/stateful_set_utils.go

```diff
@@ -165,12 +172,12 @@ func claimOwnerMatchesSetAndPod(claim *v1.PersistentVolumeClaim, set *appsv1beta
 		if hasOwnerRef(claim, set) {
 			return false
 		}
-		podScaledDown := ord >= replicaCount || reserveOrdinals.Has(ord)
+		podScaledDown := !podInOrdinalRange(pod, set, replicaCount) || reserveOrdinals.Has(ord)
```

Comment:
if podInOrdinalRange check also include reserveOrdinals, we can just call podInOrdinalRange